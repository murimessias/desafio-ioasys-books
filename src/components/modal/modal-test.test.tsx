import { screen } from '@testing-library/react'
import { renderWithTheme } from 'resources/utils/tests/helpers'

import { Modal } from '.'

describe('<Modal />', () => {
  it('should render the modal', () => {
    const { container } = renderWithTheme(<Modal isOpen>Modal</Modal>)

    expect(screen.getByLabelText('modal')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
