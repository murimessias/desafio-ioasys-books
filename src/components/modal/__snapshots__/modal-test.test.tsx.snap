// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render the modal 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgba(51,51,51,0.9);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  left: 0;
  position: fixed;
  top: 0;
  -webkit-transition: opacity 250ms ease-in;
  transition: opacity 250ms ease-in;
  width: 100%;
  z-index: 40;
  opacity: 1;
}

.c1 {
  max-width: 72rem;
  margin: 2.4rem;
  margin-top: 24rem;
  width: 100%;
}

@media (max-width:1440px) {
  .c0 {
    overflow-x: hidden;
  }
}

@media (min-width:768px) {
  .c1 {
    margin-top: 0;
  }
}

<div
  aria-hidden="false"
  aria-label="modal"
  class="c0"
>
  <div
    class="c1"
  >
    Modal
  </div>
</div>
`
